### ISSUE TITLE
Unable to Reproduce Uni-Med Results: Poor Performance on VQA, RG, and CLS Tasks Despite Matching Paper Parameters

### ISSUE THREAD

**Issue by glory6-CYL** [tsinghua-msiip/Uni-Med#7]
Thank you very much for open-sourcing this impressive work. I have followed the datasets you provided and the hyperparameters described in your paper exactly, but the trained model consistently fails to achieve the performance reported in the paper. How should I modify my experimental parameters to reach the results mentioned in your paper?
I used the following parameters as per your paper:
YAML# llm decoder
llm_model_name: "llama2"
sft_type: "lora" # lora, full, none
lora_target_modules: ['all'] # all / ['q_proj', 'k_proj', 'v_proj', 'o_proj', 'down_proj','up_proj', 'gate_proj']
lora_r: 8
lora_alpha: 16
chat_template: True
max_txt_len: 1024
end_sym: "</s>"

run:
  task: image_text_pretrain
  # optimizer
  lr_sched: "linear_warmup_cosine_lr"
  init_lr: 1e-6
  min_lr: 1e-7
  warmup_lr: 1e-6
  weight_decay: 0.05
  max_epoch: 1
  num_workers: 6
  warmup_steps: 10000
  iters_per_epoch: 100000
With these parameters, the results are significantly worse than those in the paper. For example, on the Path-VQA dataset, the BLEU score is 0.49989; on Slake-VQA_en, BLEU-1 is 0.74422; and on MedMNIST_2D_Derma, the accuracy is 0.6484 and 0.675 on organs—all lower than the reported experimental results.
Next, I trained using the parameters from your GitHub repository, with partial settings as follows:
JSON{
    "lr_sched": "linear_warmup_cosine_lr",
    "init_lr": 1e-05,
    "min_lr": 1e-07,
    "warmup_lr": 1e-06,
    "weight_decay": 0.05,
    "max_epoch": 1,
    "num_workers": 6,
    "warmup_steps": 5000,
    "iters_per_epoch": 50000,
    "seed": 42,
    "lora_r": 8,
    "lora_alpha": 16,
    "model_type": "pretrain",
    "visual_model_path": "~/.cache/lavis/eva_vit_g.pth",
    "resample_rate": 4,
    "resample_method": "projection",
    "has_moe": true,
    "projector_type": "moe_mlp",
    "num_experts": 5,
    "router_method": "router_task_token",
    "num_task_tokens": 1,
    "task_token_c": 5632,
    "router_type": "soft",
    "sft_type": "lora",
    "chat_template": true,
    "max_txt_len": 1024,
    "end_sym": "</s>"
}
With this configuration, the performance on VQA tasks is similar to the paper's results, but the report generation task performs poorly. On the MIMIC dataset, the test results are as follows:
JSON{
    "bleu_scores": {
        "bleu1": 0.2011171744436433,
        "bleu2": 0.11783784410825901,
        "bleu3": 0.07715303208225181,
        "bleu4": 0.05373860197739866
    },
    "f1_score": 0.2926497774214226,
    "meteor_scores": 0.07546323565663152,
    "rouge_scores": {
        "rouge_1": 0.2529446904819581,
        "rouge_2": 0.08614162778124669,
        "rouge_l": 0.1995193734340457
    }
}
And on the medmnist_2d_derma dataset, the accuracy is 0.712. On the dataset about the organs, the accuracy is 0.733. All are lower than the result of the paper.
I am using the exact datasets you provided—why is there such a large discrepancy? How can I adjust the experimental settings to achieve the results reported in your paper?
My training loss curve is attached (it starts high around 15, drops sharply to about 2 within the first 10k steps, and then oscillates in a low range around 0-2 for the remaining iterations up to 100k). Is this oscillation normal?
If you need to confirm any other training parameters from my setup, please let me know for further discussion. Thank you very much!<img width="591" height="295" alt="Image" src="https://github.com/user-attachments/assets/071093ce-606f-4a14-9b12-5913dee2113f" />
